lib/core/request.js

```diff
@@ -416,65 +410,41 @@ function processHeader (request, key, val, skipAppend = false) {
     return
   }
 
-  if (
-    request.host === null &&
-    key.length === 4 &&
-    key.toLowerCase() === 'host'
-  ) {
+  let headerName = headerNameLowerCasedRecord[key]
+
+  if (headerName === undefined) {
+    headerName = key.toLowerCase()
+    if (headerNameLowerCasedRecord[headerName] === undefined && !util.isValidHTTPToken(headerName)) {
```

Comment:
what key is `__proto__`?

Reply:
@KhafraDev 
This prototype is null, so it's safe. 
https://github.com/nodejs/undici/blob/8422aa988243fb4c6c37b78519954d7337cc240b/lib/core/constants.js#L113

Reply:
no problem.
```shell
> const constants = require("./lib/core/constants.js")
undefined
> constants.headerNameLowerCasedRecord.__proto__
undefined
```

Reply:
@KhafraDev 
no problem.
```shell
> $ node
Welcome to Node.js v18.19.0.
Type ".help" for more information.
> const constants = require("./lib/core/constants.js")
undefined
> constants.headerNameLowerCasedRecord.__proto__
undefined
```